skits/feature_extraction.py

```diff
@@ -69,17 +69,30 @@ def transform(self, X, y=None):
         Xt2 = self.ar2.transform(X)
         return Xt1 - Xt2
 
+class MovingAverageTransformer(BaseEstimator, TransformerMixin):
+    def __init__(self, window=5):
+        super().__init__()
+        self.window = window
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X, y=None):
+        z2 = np.cumsum(np.pad(X, ((self.window,0),(0,0)), 'constant', constant_values=0), axis=0)
+        z1 = np.cumsum(np.pad(X, ((0,self.window),(0,0)), 'constant', constant_values=X[-1]), axis=0)
+        return (z1 - z2)[(self.window-1):-1]/self.window
```

Comment:
I'm not sure if this is correct. Can you add tests so that I can see what behavior you would expect for this transformer?

Also, for initial values in `X` that are inside the `window`, can you make these `np.nan`?

Reply:
So this rolling mean gives an output the same length as the input by extending the input by as much as is necessary, eg:

rollingmean([1,2,3,4,5], window=2) would give [1.5,2.5,3.5,4.5,5]. Another approach is to change the window length dynamically as you run into the end, but this is more complicated and in most cases makes very little difference. 

I will add tests.

Reply:
done!